core/embed/rust/src/ui/model_mercury/component/prompt_screen.rs

```diff
@@ -1,76 +1,53 @@
-use crate::{
-    time::Duration,
-    ui::{
-        component::{Component, Event, EventCtx},
-        display::Color,
-        geometry::{Alignment2D, Offset, Rect},
-        shape,
-        shape::Renderer,
-    },
+use crate::ui::{
+    component::{Component, Event, EventCtx},
+    geometry::Rect,
+    model_mercury::theme,
+    shape::Renderer,
 };
 
-use super::{theme, Button, ButtonContent, ButtonMsg};
-
-const HOLD_DURATION_MS: u32 = 1000;
-const BUTTON_SIZE: i16 = 110;
+use super::{HoldToConfirm, TapToConfirm};
 
 /// Component requesting an action from a user. Most typically embedded as a
 /// content of a Frame and promptin "Tap to confirm" or "Hold to XYZ".
 #[derive(Clone)]
 pub struct PromptScreen {
     area: Rect,
-    button: Button,
-    circle_color: Color,
-    circle_pad_color: Color,
-    circle_inner_color: Color,
-    dismiss_type: DismissType,
-}
-
-#[derive(Clone)]
-enum DismissType {
-    Tap,
-    Hold,
+    tap_to_confirm: Option<TapToConfirm>,
+    hold_to_confirm: Option<HoldToConfirm>,
```

Comment:
would it be better to keep the enum with associated types? Something like
```rust
enum PromptType {
  Tap(TapToConfirm),
  Hold(HoldToConfirm),
}
```
This way the invariant is ensured and usage can be done in `match` clause

Reply:
changed https://github.com/trezor/trezor-firmware/commit/c5abc8ad08b7f65bf22f7129a701b38c3c799b1c

Reply:
and https://github.com/trezor/trezor-firmware/commit/943eba2c1d5696a9497f6a2d25e3dacdd2ffc60e